### ISSUE TITLE
MagicMirror² v2.1.0 no Modules

### ISSUE THREAD

**Issue by lolobyte** [Jopyth/MMM-Remote-Control#14]
when i choose "edit modules" the MMM-Remote-Control does't show any module.

It mean there are no modules loaded. But more than 6 modules are running at the mirror.

Any ideas?

**Comment by ejgenius** [comment]
I have the same problem :(

**Comment by jimga150** [comment]
Same here--i got a little progress by going into the node_helper.js file and changing line 63 (in the `callAfterUpdate` function):

    callAfterUpdate: function(callback, timeout) {
        if (timeout === undefined) {
            timeout = 3000;
        }

becomes:

    callAfterUpdate: function(callback, timeout) {
        if (timeout === undefined) {
            timeout = 5000; //<--here
        }

This way, the `UPDATE` socket notification is called, but the `Log.log()` function doesn't work in the non-helper js files so i can't see what happens when it gets to the `MMM-Remote-Control.js` file.

**Comment by jimga150** [comment]
I found it!! 

Okay so in version 2.1.0 of Magic Mirror, MichMich added a `disabled` option that can be applied to any module in the config file. In the MMM-Remote-Control.js file, the `sendCurrentData` function fetches ALL modules using `MM.getModules()` (as one should). Then, it loops through each one and constructs a custom JSON structure to store the properties of each module that we care about: name, position, hidden, etc.

The issue arises when it arrives at a module that is `disabled`. For some reason, the loop stops in its tracks, never going a step further than that module. it never reaches the next one, and never returns any of the modules because it's still at the `for` loop. 3 seconds pass, and the `node_helper.js` file times out, and proceeds with or without the module list (read: without.)

The solution is to comment out any module assignments in the config file if you set them to `disabled` and to add `disabled: False` to all the other modules. Then it will finish the loop, send the module list, and the remote works quite merrily.

This is the scope of what you need to comment out for a given module:
```javascript
    /*
    {
            module: 'newsfeed',
            disabled: true,
            position: 'bottom_bar',
            config: {
                feeds: [
                    	{
                        	title: "New York Times",
                        	url: "http://www.nytimes.com/services/xml/rss/nyt/H"
                    	}
                ],
                showSourceTitle: true,
                showPublishDate: true
            }
     },
    */
```
I don't know how one would go about fixing this in the code to avoid having to use this workaround, though. The whole thing puzzles me.

**Comment by Jopyth** [comment]
I will hopefully soon be able to check on all the issues with the new MM version.

**Comment by The1337Stick** [comment]
I ran in to this issue as well.  I attempted to enter the `disabled: False` entry in the rest of the modules in config.js but I just got an error that it couldn't load the config file. @jimga150 can you show a snippet of your file and how you got it working again?

**Comment by Jopyth** [comment]
Upgraded my MM version, however I did not have the time to address the issues, yet.

@The1337Stick I think this is what he did. Maybe syntax errors? Check it e.g. here: http://www.jslint.com/

**Comment by Burner911** [comment]
Thank you @jimga150 for your help! I have this issue as well and the remote not being able to hide/show modules. Have you had a chance to take a look yet? @Jopyth

**Comment by Jopyth** [comment]
Thank you guys for your patience, this should be fixed in version `1.1.0`.
